### PR TITLE
`Development`: Add docker compose files for new test servers

### PR DESCRIPTION
**Pull request by Hialus**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

We want to have *all* docker compose files in the main Artemis repository. All config should be purely done via environment variables.

### Description

This PR restructures the existing docker compose files slightly to support two new files specifically for the test servers. These new files take in a bunch of environment variables for configuration, but also offer defaults for local execution (if you so desire).
These changes are heavily related to ls1intum/artemis-ansible-collection#34.

### Steps for Testing

Not much you can do unless you want to setup a VM with ansible. Though I tested it on one of the new test servers and it works fine.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

**Review by 4ludwig4 — APPROVED** [review]
Code lgtm.
I readded and modified the comment about the system environment usage for configuration overrides in containerized artemis setups.
Concerning the UID/GID change inside the Artemis image I hope this is alright for people already using our Artemis Image (see review comment).

File: docker/artemis/Dockerfile [hunk 1]
```diff
@@ -54,6 +54,9 @@ FROM docker.io/library/eclipse-temurin:17-jdk as runtime
 
 #default path of the built .war files
 ARG WAR_FILE_PATH="/opt/artemis/build/libs"
+#default UID/GID of the artemis user
+ARG UID=1337
+ARG GID=1337
```
[the remark this answers is not in the document]
Comment (b-fein) [reply docker/artemis/Dockerfile:59]: That would indeed break our system without additional changes to the file permissions as I use rootless podman and subuid/subgid mappings.

Thanks for the heads-up.

File: docker/test-server-mysql.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    artemis-app:
+        extends:
+            file: ./artemis.yml
+            service: artemis-app
+        image: ghcr.io/ls1intum/artemis:${ARTEMIS_DOCKER_TAG:-latest}
+        depends_on:
+            mysql:
+                condition: service_healthy
+        pull_policy: always
+        restart: always
```
[the remark this answers is not in the document]
Comment (Hialus) [reply docker/test-server-mysql.yml:17]: With the early versions of your new docker compose files, I had some problems with failure:3, which is why I changed it on the test servers. Should be fine to be changed back now. WIll test it out asap.

File: docker/test-server-mysql.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# ----------------------------------------------------------------------------------------------------------------------
+# It is designed to take in a lot of environment variables to take in all the configuration of the test server.
+# ----------------------------------------------------------------------------------------------------------------------
+
+services:
+    artemis-app:
+        extends:
+            file: ./artemis.yml
+            service: artemis-app
+        image: ghcr.io/ls1intum/artemis:${ARTEMIS_DOCKER_TAG:-latest}
```
Comment (4ludwig4) [inline docker/test-server-mysql.yml:12]: Is the environment variable ``${ARTEMIS_DOCKER_TAG:-latest}`` in sync with [this line inside the ansible PR](https://github.com/ls1intum/artemis-ansible-collection/pull/34/files#diff-eae1f3b0a6e12a43a32d479475595d15fce91ae52f2c588d141c519888e0c618R33)?

Comment (Hialus) [reply docker/test-server-mysql.yml:12]: The line in the ansible collection was not correctly updated. Fixed now.

File: docker/artemis/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (dfuchss) [reply docker/artemis/Dockerfile:59]: Why do you not simply default to 1000? This is the default uid/gid for the existing installations (at least that would be the ids I would expect.

File: .dockerignore [hunk 4]
```diff
@@ -28,6 +28,13 @@ node_modules
 build/*
 !build/libs/*.war
 
+#########################
+# exclude Artemis configs
+#########################
+/src/main/resources/config/application-local*.yml
+/src/main/resources/id_*
+/src/main/resources/known_hosts
```
Comment (dfuchss) [inline .dockerignore:36]: Is the beginning slash correct?

File: docker/artemis/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hialus) [reply docker/artemis/Dockerfile:59]: Using UID/GID 1000 is sadly not an option for us. These are already taken by another user on every VM that we afaik cannot simply change.
All existing Artemis instances use 1337 instead, which is also the default in the artemis-ansible-collection. So we also thought we would use it as the default here as well.

While we could default to 1000 in the Dockerfile, we would then build the image using 1337. So in any case to use 1000 you would have to build a separate image.

File: .dockerignore (hunk 4 above)
Comment (4ludwig4) [reply .dockerignore:36]: It should be fine [according to the docs and it's examples](https://docs.docker.com/engine/reference/builder/#dockerignore-file).

**Review by Mtze — APPROVED** [review]

**Pull request merged**